src/context/FeatureFlagContext.tsx

```diff
@@ -0,0 +1,86 @@
+import { useRouter } from "next/router";
+import { ParsedUrlQuery } from "querystring";
+import React, { createContext, useContext, useEffect, useState } from "react";
+import { extractQueryParam } from "../util/LinkUtils";
+
+type FeatureFlag = {
+  [flag: string]: boolean;
+};
+
+type FeatureFlagContextType = {
+  featureFlags: FeatureFlag;
+  setFeatureFlags: (featureFlags: FeatureFlag) => void;
+  isFlagActive: (flag: string) => boolean;
+};
+
+export const FeatureFlagContext =
+  createContext<FeatureFlagContextType>(undefined);
+
+const extractFeatureFlagParams = (query: ParsedUrlQuery) => {
+  const featureFlags = {};
+  for (const param in query) {
+    if (param.includes("feature_")) {
+      const featureFlag = param.split("_")[1];
+      featureFlags[featureFlag] = JSON.parse(extractQueryParam(query, param));
+    }
+  }
+  return featureFlags;
+};
+
+export const FeatureFlagProvider: React.FC = ({ children }) => {
+  const [featureFlags, setFeatureFlags] = useState<FeatureFlag>({});
+  const isFlagActive = (flag: string) => {
+    return featureFlags[flag];
+  };
+
+  const router = useRouter();
+  const { query } = router;
+
+  useEffect(() => {
+    const storedFeatureFlagsStr = sessionStorage.getItem("featureFlags");
```

Comment:
Can you explain how the functionality interacts with the session storage?  It seems from the tests that it's all based on what the query parameter is, so why is it looking into session storage at all?